utils/manifest-utils/src/matching/domain.rs

```diff
@@ -0,0 +1,143 @@
+use std::{
+    collections::HashSet,
+    convert::TryFrom,
+    fs::{self, File},
+    io::{self, BufReader},
+    path::{Path, PathBuf},
+    sync::{Arc, Mutex, RwLock},
+};
+
+use regex::RegexSetBuilder;
+use serde::{Deserialize, Serialize};
+use ya_utils_path::SwapSave;
+
+use super::{CompositeMatcher, Matcher, RegexMatcher, StrictMatcher};
+use crate::{util::str_to_short_hash, ArgMatch};
+
+pub type DomainsMatcher = CompositeMatcher;
+pub type SharedDomainPatterns = Arc<Mutex<DomainPatterns>>;
+pub type SharedDomainMatchers = Arc<RwLock<DomainsMatcher>>;
+
+#[derive(Clone, Default, Debug)]
+pub struct DomainWhitelistState {
+    pub patterns: SharedDomainPatterns,
+    pub matchers: SharedDomainMatchers,
+}
+
+impl TryFrom<DomainPatterns> for DomainWhitelistState {
+    type Error = anyhow::Error;
+
+    fn try_from(patterns: DomainPatterns) -> Result<Self, Self::Error> {
+        let matcher = DomainsMatcher::try_from(&patterns)?;
+        let matcher = Arc::new(RwLock::new(matcher));
+        let patterns = Arc::new(Mutex::new(patterns));
+        Ok(Self {
+            patterns,
+            matchers: matcher,
+        })
+    }
+}
+
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+pub struct DomainPatterns {
+    pub patterns: Vec<DomainPattern>,
+}
+
+impl TryFrom<&PathBuf> for DomainPatterns {
+    type Error = anyhow::Error;
+
+    fn try_from(path: &PathBuf) -> Result<Self, anyhow::Error> {
+        let file = File::open(path)?;
+        let reader = BufReader::new(file);
+        Ok(serde_json::from_reader(reader)?)
+    }
+}
```

Comment:
I don't think this matches the semantics of `TryFrom`. (Try)From encode (fallible) type conversions, where you intend to keep the same data but in a different form, like `2i32` -> `2.0f32`. It also seems to partly duplicate the logic from `DomainPatterns::load`, is there a reason for that?

Reply:
At some point I started using `From`/`TryFrom` as a default trait for a constructor.
I was not even aware it is in `core::convert`. Thanks for this one. I removed some other case of incorrect `impl TryFrom`, but probably there are others.
Done.